app/src/main/java/chat/rocket/android/contacts/adapter/ContactRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,96 @@
+package chat.rocket.android.contacts.adapter
+
+import android.content.Intent
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import chat.rocket.android.R
+import chat.rocket.android.chatrooms.adapter.*
+import chat.rocket.android.main.ui.MainActivity
+import chat.rocket.android.util.extensions.inflate
+import kotlin.collections.HashMap
+
+
+class ContactRecyclerViewAdapter(
+        private val context: MainActivity,
+        private val contactArrayList: List<ItemHolder<*>>
+
+
+) : RecyclerView.Adapter<ViewHolder<*>>() {
+
+    init {
+        setHasStableIds(true)
+    }
+
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder<*> {
+        return when (viewType) {
+            VIEW_TYPE_CONTACT -> {
+                val view = parent.inflate(R.layout.item_contact)
+                ContactViewHolder(view)
+            }
+            VIEW_TYPE_HEADER -> {
+                val view = parent.inflate(R.layout.item_heading)
+                ContactHeaderViewHolder(view)
+            }
+            VIEW_TYPE_INVITE -> {
+                val view = parent.inflate(R.layout.item_invite)
+                InviteViewHolder(view)
+            }
+            else -> throw IllegalStateException("View type must be either Room, Header or Invite")
+        }
+    }
+
+    override fun getItemCount() = contactArrayList.size
+
+    override fun getItemId(position: Int): Long {
+        val item = contactArrayList[position]
+        return when (item) {
+            is ContactItemHolder -> item.data.hashCode().toLong()
+            is ContactHeaderItemHolder -> item.data.hashCode().toLong()
+            is inviteItemHolder -> item.data.hashCode().toLong()
+            else -> throw IllegalStateException("View type must be either Room, Header or Invite")
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return when (contactArrayList[position]) {
+            is ContactItemHolder -> VIEW_TYPE_CONTACT
+            is ContactHeaderItemHolder -> VIEW_TYPE_HEADER
+            is inviteItemHolder -> VIEW_TYPE_INVITE
+            else -> throw IllegalStateException("View type must be either Room, Header or Invite")
+        }
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder<*>, position: Int) {
+        if (holder is ContactViewHolder) {
+            holder.bind(contactArrayList[position] as ContactItemHolder)
+            holder.itemView.setOnClickListener {
```

Comment:
I'm confused by this.......Is your intention to setOnClickListener at this point, or inside ContactViewHolder.kt?